lib/Doctrine/DBAL/Migrations/Tools/Console/Command/DiffCommand.php

```diff
@@ -109,7 +109,7 @@ public function execute(InputInterface $input, OutputInterface $output)
         }
 
         $version = date('YmdHis');
-        $path = $this->generateMigration($configuration, $input, $version, $up, $down);
+        $path = $this->generateMigration($configuration, $input, $version, $up, $down, true);
```

Comment:
The concerns of Martin and Fowler apply when a routine does completely different things based on a flag:

```php
function doStuff($flag)
{
  if ($flag) {
    doSomething();
  } else {
    doSomethingCompleteDifferent();
  }
}
```

The method in question does exactly one thing (generate migrations) and the flag is appropriate for configuration. It can be made more reasonable with a constant if it's really a concern.

```php
$this->generateMigration(..., self::SQL_ONLY);
```